### PR TITLE
Drop-down menu to list all families in create placeholder

### PR DESCRIPTION
**Pull request by friquette**
## Changelog Description
Currently in the create placeholder window, we need to write the family manually. This replace the text field by an enum field with all families for the current software.

## Additional info
For Maya placeholder window:
image

For Nuke placeholder window:
image

### CODE REVIEW

**Comment by iLLiCiTiT** [comment]
Since it is from "free text" to "explicit value", looking for families to Nuke creator plugins is limited as loaders can usually use other families that cannot be produced by host creators. So I would recommend to rather use `families` attribute from load plugins instead of create plugins...

File: openpype/pipeline/workfile/workfile_template_builder.py [hunk 1]
```diff
@@ -1246,6 +1249,18 @@ def get_load_plugin_options(self, options=None):
...
+        # Get families and sort them alphabetically
+        all_loaders = discover_loader_plugins()
+        families = []
+        for loader in all_loaders:
+            families.extend(loader.families)
+
+        families = list(set(families))
+        if "*" in families:
+            families.remove("*")
+        families.sort()
```
Comment (BigRoy) [inline openpype/pipeline/workfile/workfile_template_builder.py:1262]: How about this?

```suggestion
		# Get combined families from all loaders excluding "*"
        families = set()
        for loader in all_loaders:
            families.update(loader.families)
		families.discard("*")
		
		# Sort for readability
		families = list(sorted(families))
```

File: openpype/pipeline/workfile/workfile_template_builder.py [hunk 2]
```diff
@@ -43,8 +44,10 @@
     get_contexts_for_repre_docs,
     load_with_repre_context,
 )
+
 from openpype.pipeline.create import (
     discover_legacy_creator_plugins,
+    discover_creator_plugins,
```
Comment (BigRoy) [inline openpype/pipeline/workfile/workfile_template_builder.py:50]: This appears to be unused?
```suggestion
```

File: openpype/pipeline/workfile/workfile_template_builder.py (hunk 1 above)
Comment (iLLiCiTiT) [reply openpype/pipeline/workfile/workfile_template_builder.py:1262]: I like this more.

File: openpype/pipeline/workfile/workfile_template_builder.py [hunk 3]
```diff
@@ -1246,6 +1249,18 @@ def get_load_plugin_options(self, options=None):
 
         loader_items = list(sorted(loader_items, key=lambda i: i["label"]))
         options = options or {}
+
+        # Get families and sort them alphabetically
+        all_loaders = discover_loader_plugins()
+        families = []
```
[the remark this answers is not in the document]
Comment (friquette) [reply openpype/pipeline/workfile/workfile_template_builder.py:1255]: I can do it with a list comprehension, but honestly I don't know if it is really necessary. Are the 1 or 2 lines we gain here worth the lost in readability ?

[the remark this answers is not in the document]
Comment (iLLiCiTiT) [reply openpype/pipeline/workfile/workfile_template_builder.py]: I don't think this is needed with https://github.com/ynput/OpenPype/pull/4928#discussion_r1187602670

**Comment by iLLiCiTiT** [comment]
Was possible to "not fill" the family input?

**Comment by friquette** [comment]
> Was possible to "not fill" the family input?

Like an empty entry ? That maybe we can pass by default ?

**Comment by iLLiCiTiT** [comment]
> Like an empty entry ? That maybe we can pass by default ?

It was meant as what happens if the family is empty? Would it work (somehow?). If yes, we should add empty string to enum, if it would crash then this is better.

**Comment by friquette** [comment]
> > Like an empty entry ? That maybe we can pass by default ?
> 
> It was meant as what happens if the family is empty? Would it work (somehow?). If yes, we should add empty string to enum, if it would crash then this is better.

The family and representations are mandatory. If one is missing, the workfile builder will not find any asset

**Comment by friquette** [comment]
> > Like an empty entry ? That maybe we can pass by default ?
> 
> It was meant as what happens if the family is empty? Would it work (somehow?). If yes, we should add empty string to enum, if it would crash then this is better.

The family and representations are mandatory. If one is missing, the workfile builder will not find any asset

File: openpype/pipeline/workfile/workfile_template_builder.py [hunk 4]
```diff
@@ -24,6 +24,7 @@
     get_linked_assets,
     get_representations,
 )
+from openpype.pipeline.load.plugins import discover_loader_plugins
```
Comment (iLLiCiTiT) [inline openpype/pipeline/workfile/workfile_template_builder.py:27]:
```suggestion
```

**Review by iLLiCiTiT — APPROVED** [review]
Code looks ok. Can someone validate usability?

**Review by iLLiCiTiT — APPROVED** [review]
Code looks ok. Can someone validate usability?

**Review by LiborBatek — APPROVED** [review]
All works as expected in Maya Template Builder.
Instead of writting families manually rollout containing all families is present and working fine,

Did test to create several types of families and also ```Update Placeholder``` actions too. All good!

image

**Review by Minkiu — APPROVED** [review]

**Pull request merged**